aspnetcore/signalr/authn-and-authz.md

```diff
@@ -176,6 +176,62 @@ public class ChatHub: Hub
 }
 ```
 
+::: moniker range=">= aspnetcore-3.0"
+
+### Use authorization handlers to customize hub method authorization
+
+SignalR passes a custom resource to authorization handlers when a hub method requires authorization. The resource is of type `HubInvocationContext` and contains `HubCallerContext`, the name of the hub method, and the arguments to the hub method for use in custom authorization logic.
+
+The following example shows a custom handler that only allows users with the same name as the hub method to invoke the method.
+
+```csharp
+public class HubMethodAuthorizationHandler :
+    AuthorizationHandler<HubMethodRequirement, HubInvocationContext>
+{
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+                                                   HubMethodRequirement requirement,
+                                                   HubInvocationContext resource)
+    {
+        if (context.User?.FindFirst(ClaimTypes.NameIdentifier)?.Value == resource.HubMethodName)
+        {
+            context.Succeed(requirement);
+        }
+
+        return Task.CompletedTask;
+    }
+}
+
+public class HubMethodRequirement : IAuthorizationRequirement { }
+
+public class ChatHub : Hub
+{
+    [Authorize("NamedMethod")]
+    public void Steve()
+    {
+    }
+}
+```
+
+Setup the policy and register the handler in DI.
```

Comment:
```suggestion
Set up the policy and register the handler in DI.
```

"Setup" is a noun, such as "it's a setup!". "Set up" is the verb phrase you're looking for ;).